shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp

```diff
@@ -28,12 +28,18 @@ namespace shared_model {
   namespace proto {
     class AddAssetQuantity final : public interface::AddAssetQuantity {
      private:
-      template<typename Value>
+      template <typename Value>
       using Lazy = detail::LazyInitializer<Value>;
 
      public:
       explicit AddAssetQuantity(const iroha::protocol::Command &command)
-          : AddAssetQuantity(command.add_asset_quantity()) {}
+          : AddAssetQuantity(command.add_asset_quantity()) {
+        if (not command.has_add_asset_quantity()) {
+          // TODO 11/11/17 andrei create generic exception message
+          throw std::invalid_argument(
```

Comment:
I don't really sure it's a good idea, factory that returns optional is definitely better

Reply:
for now it's possible to use boost_assert